Sources/Common/DataModel/Polygon/index.js

```diff
@@ -174,45 +193,629 @@ export function getBounds(poly, points, bounds) {
 
 // ---------------------------------------------------
 /**
- * Compute the normal of a polygon and return its norm.
+ * Compute the normal of a polygon.
  *
- * TBD: This does the same thing as vtkPolygon.computeNormal,
- * but in a more generic way. Maybe we can keep the public
- * static method somehow and have the private method use it instead.
- *
- * @param {Array<Number>|TypedArray<Number>} poly
+ * @param {Vector3[]} poly
  * @param {vtkPoints} points
  * @param {Vector3} normal
  * @returns {Number}
  */
 export function getNormal(poly, points, normal) {
```

Comment:
let's keep the API simple, only support `vtkPoints` instead of a points/poly pair.

So you need to modify setPoints to optionally take in a list of pointIds: `setPoints(points : [[],[],[]]|[], pointIds = null|[])`